### DISCUSSION TITLE
Blurry photo detection

### DISCUSSION THREAD

**Discussion by stefanbuck** [ente-io/ente#3579] [Enhancements]
Right now, there’s no easy way to filter out photos that are blurry, and it can be a pain to manually go through them all. It would be amazing if the app could automatically detect blurry photos (maybe even bad lighting etc.) and group them together, so I can quickly review and delete them in bulk

**Comment by eric-gitta-moore** [comment]
It is indeed a very practical function, but what technology can be judged at present. I do know that there seem to be models that can identify the clarity of different images in the same scene.

**Reply by stefanbuck** [reply]
This can be accomplished with math like describe here https://pyimagesearch.com/2015/09/07/blur-detection-with-opencv/ 

No need to run expensive models locally or in the cloud

**Comment by reflector8939** [comment]
Immich already does this, it'd be great for ente too
